### ISSUE TITLE
False positive on MissingUseCall when assigning a ExecutorCoroutineDispatcher to a property

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#8600]
## Expected Behavior
This shouldn't be a violation

## Observed Behavior
It is 

## Steps to Reproduce
```
override val read: CoroutineDispatcher = newFixedThreadPoolContext(4, "db-reads")
```

## Your Environment
* Version of detekt used: 2.0.0-alpha.0
* Version of Gradle used (if applicable): 9.0.0

**Comment by eygraber** [comment]
Similar scenario where the Closeable is returned from a lambda

```kotlin
@Provides
  @SingleIn(AppScope::class)
  fun provideSqlDelightDriver(
    @AppContext context: Context,
  ): AndroidxSqliteDriver = AndroidxSqliteDriver(
    createConnection = { name ->
      BundledSQLiteDriver().open(fileName = name)
    },
    databaseType = AndroidxSqliteDatabaseType.FileProvider(
      context = context,
      name = "cs3.db",
    ),
    schema = MyDb.Schema,
    configuration = AndroidxSqliteConfiguration(
      isForeignKeyConstraintsEnabled = true,
      journalMode = SqliteJournalMode.WAL,
    ),
  )
```

**Comment by eygraber** [comment]
And another where it gets returned from a function:

```kotlin
@Inject
class CacheBusterInterceptor(
  private val requestHelper:  RequestHelper,
) : Interceptor {
  @OptIn(ExperimentalUuidApi::class)
  override fun intercept(chain: Interceptor.Chain): Response {
    val request = chain.request()
    val isCacheBusterRequired = request.header(SKIP_CLOUDFRONT_CACHE)?.toBooleanStrictOrNull() != false

    val newRequest = when {
      requestHelper.isAuthRequest(request) -> request

      else ->
        request
          .newBuilder()
          .removeHeader(SKIP_CLOUDFRONT_CACHE)
          .run {
            when {
              isCacheBusterRequired -> header("Cache-Buster", Uuid.random().toHexDashString())
              else -> this
            }
          }
          .build()
    }

    return chain.proceed(newRequest)
  }
}
```

**Comment by eygraber** [comment]
Bump
